File: src/main/java/com/nayasis/simplelauncher/service/terminal/helper/FxHelper.java
package com.nayasis.simplelauncher.service.terminal.helper;

import io.nayasis.common.basicafx.javafx.etc.FxThread;
import javafx.scene.control.Alert;
import javafx.scene.control.ButtonType;
import javafx.scene.control.TextInputDialog;
import javafx.scene.paint.Color;

import java.util.Optional;
import java.util.concurrent.CompletableFuture;

/**
 * @see <a>https://github.com/javaterminal/TerminalFX</a>
 * Created by usta on 26.09.2016.
 */
public class FxHelper {

    public static String colorToHex(Color color) {
        return String.format("#%02X%02X%02X",
                (int) (color.getRed() * 255),
                (int) (color.getGreen() * 255),
                (int) (color.getBlue() * 255));
    }

    public static boolean askQuestion(String message) {
        CompletableFuture<Boolean> completableFuture = new CompletableFuture<Boolean>();
        completableFuture.runAsync(() -> FxThread.runLater(() -> {
            Alert alert = new Alert(Alert.AlertType.INFORMATION, message, ButtonType.YES, ButtonType.NO);
            ButtonType buttonType = alert.showAndWait().orElse(ButtonType.NO);
            completableFuture.complete(buttonType == ButtonType.YES);
        }) );

        return completableFuture.join();
    }

    public static String askInput(String message) {
        CompletableFuture<String> completableFuture = new CompletableFuture<>();
        completableFuture.runAsync(() -> {
            FxThread.runLater(() -> {
                TextInputDialog inputDialog = new TextInputDialog();
                inputDialog.setContentText(message);
                Optional<String> optional = inputDialog.showAndWait();
                completableFuture.complete(optional.orElse( null ));
            });
        });

        return completableFuture.join();
    }
}

File: src/main/kotlin/com/github/nayasis/simplelauncher/view/terminal/FxHelper.kt
package com.github.nayasis.simplelauncher.view.terminal

import javafx.application.Platform
import javafx.scene.control.Alert
import javafx.scene.control.ButtonType
import javafx.scene.control.TextInputDialog
import javafx.scene.paint.Color
import java.util.concurrent.CompletableFuture

object FxHelper {

    fun askQuestion(message: String?): Boolean {
        val completableFuture = CompletableFuture<Boolean>()
        CompletableFuture.runAsync {
            Platform.runLater {
                val alert = Alert(
                    Alert.AlertType.INFORMATION,
                    message,
                    ButtonType.YES,
                    ButtonType.NO
                )
                val buttonType = alert.showAndWait().orElse(ButtonType.NO)
                completableFuture.complete(buttonType == ButtonType.YES)
            }
        }
        return completableFuture.join()
    }

    fun askInput(message: String?): String? {
        val completableFuture = CompletableFuture<String?>()
        CompletableFuture.runAsync {
            Platform.runLater {
                val inputDialog = TextInputDialog()
                inputDialog.contentText = message
                val optional = inputDialog.showAndWait()
                completableFuture.complete(optional.orElse(null))
            }
        }
        return completableFuture.join()
    }
}


fun Color.toHex(): String {
    return "#%02X%02X%02X".format(
        (red * 255).toInt(),
        (green * 255).toInt(),
        (blue * 255).toInt()
    )
}